src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -557,8 +557,17 @@ private static Exception CreateHttpRequestException(Exception inner = null)
             return new HttpRequestException(SR.net_http_client_execution_error, inner);
         }
 
+        private static void ResponseFormatAssert(bool condition)
+        {
+            if (!condition)
+            {
+                throw new HttpRequestException(SR.net_http_unix_invalid_response);
```

Comment:
Sorry @kapilash, I didn't realize that CreateHttp.. uses a generic message. Change looks fine to me